### ISSUE TITLE
Extra new line added

### ISSUE THREAD

**Issue by IvanGoncharov** [nodeca/js-yaml#221]
Test:
```js
var YAML = require('js-yaml');

data = "test1\ntest2"
console.log(data);
console.log('===============================================================');
data = YAML.safeDump(data);
console.log(data);
console.log('===============================================================');
data = YAML.safeLoad(data);
console.log(data);
console.log('===============================================================');
```
Output:
```
$ node test.js 
test1
test2
===============================================================
|-
test1
test2

===============================================================
test1
test2

===============================================================
```
So extra new line added in the end of string.

**Comment by puzrin** [comment]
Dupe?

**Comment by IvanGoncharov** [comment]
@puzrin If you ask about #222, then no. It's about trailing new line and not related to 80 symbols.

**Comment by puzrin** [comment]
Probably, that's because of #125 workaround (the same exists in Ruby's parser).

**Comment by dervus** [comment]
Both this and #222 actually caused by combination of two different bugs: one in the loader, and another in the dumper.

Loader bug:
- It doesn't throw an error on block scalar without indentation. Block scalars **cannot** have 0-level indentation.

Dumper bug:
- It doesn't add indentation for top-level block scalar.

**Comment by dervus** [comment]
Dumper part if fixed by #274.

I've created separate issue for the loader bug: https://github.com/nodeca/js-yaml/issues/280
